### DISCUSSION TITLE
Doesn't work on big sur

### DISCUSSION THREAD

**Discussion by rririanto** [noah-nuebling/mac-mouse-fix#64] [Questions] [answered]
Hi, I use mac mouse fix since the beginning. but I don't know why after upgrading to big sur it doesn't work correctly. 

I mean it was successfully installed but when I click to enable it doesn't work and it doesn't produce any error message. 

Does anyone experience the same things? 

<img width="672" alt="Screen Shot 2021-03-06 at 18 17 09" src="https://user-images.githubusercontent.com/935878/110205339-00545900-7eaa-11eb-8f4d-8da59d63e456.png">

**Comment by rririanto** [comment]
I don't know what step I miss, but it works right now. A few things I did:

- I make sure that it's enabled on privacy
<img width="355" alt="Screen Shot 2021-03-06 at 19 12 08" src="https://user-images.githubusercontent.com/935878/110206391-e74ea680-7eaf-11eb-8b02-713e4da8d14d.png">

- I moved the extracted file to applications
<img width="774" alt="Screen Shot 2021-03-06 at 19 13 07" src="https://user-images.githubusercontent.com/935878/110206410-09482900-7eb0-11eb-82c9-5ac075b51f83.png">


- I tried to reinstall the app 

If I can give advice, maybe we can have an installation process where tells the user to move the Mac Mouse Fix to applications then a dialog to let the user move the installation file to the trash. 

Anyway, Thank you for making this the most useful apps, It's really amazing! Good job! Well done!

**Reply by noah-nuebling** [reply]
Hey @rririanto thanks for your contribution and for your nice words!
macOS has some security features that make apps run in a sort of Quarantine when you run them without moving them our of the Downloads folder first (It's called "App Translocation").
I think your issues might have something to do with that, but I'm not sure. 

It would definitely be really nice, if Mac Mouse Fix could detect App Translocation and warn the user, or even move itself to the Applications folder or something similar. I'll put that on the agenda for the future. Thanks for your input!

Edit: The later 2.0.0 Betas do automatically untranslocate themselves! So this shouldn’t be an issue anymore

**Reply by rririanto** [reply]
Aha! I just know about the app "App Translocation". I tried to reinstall again and tried without move to the Applications folder and you are right it does not work. Then I tried again to clean the installations and moved the app to applications and it works perfectly. Thank you again for your help!

**Comment by njyo** [comment, marked as answer]
FYI: Anybody seeing this screen when trying to run the app (e.g. because Safari auto-unzips the file):
![2021-03-24 at 19 07](https://user-images.githubusercontent.com/988408/112301152-98d54080-8cd4-11eb-8e5c-106af255f4b0.png)

You need to go into the `General` tab in the `Security & Privacy` settings and open the app from there:
![2021-03-24 at 19 03](https://user-images.githubusercontent.com/988408/112301059-89ee8e00-8cd4-11eb-8069-0969dfb285a5.png)

**Reply by Cramertech** [reply]
I believe that generally this behavior is expected for anything downloaded from the web and not from the actual App Store. The method you showed is one way to handle it. Another might be to right-click (or ctrl+click) and select Open, then on the corresponding pop up you click Open. This is what I had to do. Also, you may not have that option (I'm not sure) until you go to your system Preferences > Security & Privacy > Allow apps downloaded from: > App Store and identified developers.

<img width="780" alt="Screen Shot 2021-04-03 at 9 20 32 AM" src="https://user-images.githubusercontent.com/2154022/113484608-0f550980-945e-11eb-8813-a04b660be5b8.png">
<img width="372" alt="lol - just used iTerm as an example since I've already allowed Mouse Fix" src="https://user-images.githubusercontent.com/2154022/113485872-bb015800-9464-11eb-98f7-caabba9cb489.png">[Just used iTerm as an example since I've already allowed Mouse Fix]

**Reply by njyo** [reply]
Thanks, for the addition, @Cramertech. Yeah, the setting for `App Store and identified developers` is crucial and I should have mentioned it. :)

However, I was stumped this time, because I was unable to get the same `Open` dialogue as you showed via `CMD+Click` » `Open`. This used to work before like a charm, but for some reason, it's not an option for me anymore. Not sure if that's Big Sur, or the M1, or something else. The only place I found the ability to enable this, was in the `General` tab for `Security & Privacy` 🤷

**Reply by noah-nuebling** [reply]
Hey guys! Marking this as an answer. As far as I can tell, the underlying problem of the original poster was the App Translocation stuff, not this. But that has been resolved in a later Mac Mouse Fix 2 Beta.

On the other hand, the problem that @njyo's answer solves is one that people may still run into, so I'm marking it as an answer for visibility. And also so that all the resolved discussions have an answer which helps me keep an overview.

**Comment by hezral** [comment]
Hi, i wanted to try this app but seems it wouldn't even open. Any ideas?
Currently on 11.3.1

![image](https://user-images.githubusercontent.com/762735/117085561-bccc7d00-ad7c-11eb-9733-ca3f0fa918c3.png)

**Reply by njyo** [reply]
Hi, do you have admin rights on your computer? :$
Not sure, why you would not see the same warnings with an ability to override.

**Reply by hezral** [reply]
i do, its the only account on it. I've never see it either. Also i'm trying the 2.0.0 Beta 6 version. 

Also can't compile it myself since it requires a developer account.

**Reply by njyo** [reply]
Ugh. That sucks. The only other thing I can think of, is as @Cramertech wrote above, to try to right-click and then select `Open`, if that triggers a different warning window…

**Reply by huszarrichard** [reply]
Same problem here. :( "Right-click -> Open" method doesn't work.

**Reply by hezral** [reply]
Ok, so there is an option to override the malware prompt and its gone after that but still says i don't have permissions to run it. 
To check, i also didn't move the .app to the Applications folder and same result

![image](https://user-images.githubusercontent.com/762735/117101351-9fada380-ada8-11eb-99fc-63ed035dcaa7.png)

**Reply by huszarrichard** [reply]
> Ok, so there is an option to override the malware prompt and its gone after that but still says i don't have permissions to run it.
> To check, i also didn't move the .app to the Applications folder and same result
> 
> ![image](https://user-images.githubusercontent.com/762735/117101351-9fada380-ada8-11eb-99fc-63ed035dcaa7.png)

I also tried, doesn't work, I don't have permissions to run it.
![CleanShot 2021-05-05 at 08 06 57](https://user-images.githubusercontent.com/17738461/117102570-e1bef100-ad78-11eb-9a8d-e80392fdcbd1.png)

**Reply by islavstan** [reply]
Hi! Today I got the same error, app worked perfectly before

**Reply by hezral** [reply]
Do you happen to be on 11.3.1 os?

**Reply by islavstan** [reply]
no, 11.2.2

**Reply by islavstan** [reply]
I didnt update os, yesterday app worker, today I got this error

**Reply by hezral** [reply]
interesting. something else then

**Reply by hezral** [reply]
https://github.com/noah-nuebling/mac-mouse-fix/issues/95#issuecomment-832613565 this fix worked for me now. enjoying the app so far!

**Reply by noah-nuebling** [reply]
Nice! Let me know if you have anymore feedback.

**Reply by hezral** [reply]
will do, and expect a milkshare coming your way 👍🏻 

edit: ok i can't donate since PayPal doesn't work in my country for donation it seems. any other way?

**Reply by noah-nuebling** [reply]
Thanks for contributing everyone!

There are now new versions of the app available which should fix these error messages:

- [1.0.4](https://github.com/noah-nuebling/mac-mouse-fix/releases/tag/1.0.4)
- [2.0.0 Beta 8](https://github.com/noah-nuebling/mac-mouse-fix/releases/tag/2.0.0-Beta-8)

I hope these work for everyone. If not be sure to let me know.

For more info on what's going on, see https://github.com/noah-nuebling/mac-mouse-fix/issues/95

**Reply by noah-nuebling** [reply]
@hezral Aw thanks! There is currently no other way to donate but I really appreciate your intention :) That means much more to me than the actual money.

**Comment by rririanto** [comment]
I faced the same issues. The last time I updated the OS version is a week ago. But the popup showed up today. And If I click cancel the dialog will always appear to ask me to move it to trash. If I click to move to trash the application will be removed and I no longer can use it.  

- OS Big Sur 11.2.3
<img width="379" alt="Screen Shot 2021-05-06 at 11 50 12" src="https://user-images.githubusercontent.com/935878/117243946-57eb5280-ae62-11eb-885a-ed4d018c0de4.png">
<img width="674" alt="Screen Shot 2021-05-06 at 11 54 33" src="https://user-images.githubusercontent.com/935878/117243971-633e7e00-ae62-11eb-878a-372edc4c0c16.png">

**Reply by hezral** [reply]
See this https://github.com/noah-nuebling/mac-mouse-fix/issues/95#issuecomment-832613565

**Reply by noah-nuebling** [reply]
Hey guys, thanks for contributing, and thanks for taking the time to attach some screenshots @rririanto.

There are now new versions of the app available which should fix these error messages:

- [1.0.4](https://github.com/noah-nuebling/mac-mouse-fix/releases/tag/1.0.4)
- [2.0.0 Beta 8](https://github.com/noah-nuebling/mac-mouse-fix/releases/tag/2.0.0-Beta-8)

I hope these work for everyone. If not be sure to let me know.

For more info on what's going on, see https://github.com/noah-nuebling/mac-mouse-fix/issues/95

**Comment by hezral** [comment]
Direct to paypal account should work i think, unless paypal is just in the
middle to process the transactions. I'll be happy to send via that way if
you have a paypal account. I' m sure there are others who are unable to do
so as well. :)

**Reply by noah-nuebling** [reply]
Yeah you're right actually, I've gotten quite a few messages from people not being able to donate due to their location so it might be worth changing the donation channel. 

I've now set up a [PayPal.me](https://paypal.me/mousefix), does that work for you? 

Oh and does it display in German for you? I can't get it to display in English in my browser.

Thanks a lot for your help with getting me them milkshakes!

**Comment by hezral** [comment]
Sent! Thanks again for making this. I hope it gets more useful features.

**Reply by noah-nuebling** [reply]
Thanks a lot! 
This is gonna be a malted milk one I think. I've never tried that but it's pretty big in the US from what I heard so I'm excited! 
I plan on adding some neat new features this summer so stay tuned! ;)

**Reply by noah-nuebling** [reply]
You can always write an [enhancement request](https://noah-nuebling.github.io/mac-mouse-fix-feedback-assistant/?type=feature-request) if you have any specific features in mind.
